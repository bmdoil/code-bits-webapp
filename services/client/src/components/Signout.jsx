import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Signout extends Component {
    componentDidMount() {
        this.props.signoutUser();
    };
    render() {
        return (
            <div>
                <p>Signed Out. <Link to="/login">Login.</Link></p>
            </div>
        )
    };
};

export default Signout;