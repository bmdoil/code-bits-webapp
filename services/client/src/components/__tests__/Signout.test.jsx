import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import Signout from '../Signout';

// Mock user to log out with
const user = jest.fn()

test('Signout renders', () => {
    const wrapper = shallow(<Signout signoutUser={ user }/>);
    const element = wrapper.find('p');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children[0]).toContain('Signed Out')
});


test('Signout renders a snapshot', () => {
    const tree = renderer.create(
      <Router><Signout signoutUser={user}/></Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });