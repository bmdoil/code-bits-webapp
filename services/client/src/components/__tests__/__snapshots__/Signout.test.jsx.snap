// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signout renders a snapshot 1`] = `
<div>
  <p>
    Signed Out. 
    <a
      href="/login"
      onClick={[Function]}
    >
      Login.
    </a>
  </p>
</div>
`;
